x/concentrated-liquidity/invariant_test.go

```diff
@@ -106,10 +119,15 @@ func (s *KeeperTestSuite) assertTotalRewardsInvariant() {
 	s.Require().Equal(initialTotalPoolLiquidity, finalTotalPoolLiquidity)
 
 	// Ensure total remaining spread rewards and incentives are exactly equal to loss due to rounding
-	roundingLossSpread := expectedTotalSpreadRewards.Sub(totalCollectedSpread)
-	s.Require().Equal(roundingLossSpread, remainingTotalSpreadRewards)
-	roundingLossIncentives := expectedTotalIncentives.Sub(totalCollectedIncentives)
-	s.Require().Equal(roundingLossIncentives, remainingTotalIncentives)
+	if expectedGlobalRewardValues.TotalSpreadRewards == nil {
+		roundingLossSpread := expectedTotalSpreadRewards.Sub(totalCollectedSpread)
+		s.Require().Equal(roundingLossSpread, remainingTotalSpreadRewards)
+	}
+
+	if expectedGlobalRewardValues.TotalIncentives == nil {
+		roundingLossIncentives := expectedTotalIncentives.Sub(totalCollectedIncentives)
+		s.Require().Equal(roundingLossIncentives, remainingTotalIncentives)
+	}
```

Comment:
it: should we have a guard to ensure this is not-nil to avoid having erroneous passes?

Reply:
Well if it's not nil that means we're asserting exact expected amounts so these branches would no longer make sense